src/pages/home/report/FloatingMessageCounter/index.js

```diff
@@ -28,73 +26,67 @@ const defaultProps = {
-                    <View style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter]}>
-                        <Button
-                            success
-                            small
-                            onPress={this.props.onClick}
-                        >
-                            <View style={[styles.flexRow, styles.alignItemsCenter]}>
-                                <Icon
-                                    small
-                                    src={Expensicons.DownArrow}
-                                    fill={themeColors.textLight}
-                                />
-                                <Text
-                                    selectable={false}
-                                    style={[styles.ml2, styles.buttonSmallText, styles.textWhite]}
-                                >
-                                    {this.props.translate('newMessages')}
-                                </Text>
-                            </View>
-                        </Button>
-                    </View>
+    useEffect(() => {
+        if (props.isActive) {
+            show();
+        } else {
+            hide();
+        }
+    }, [props.isActive, show, hide]);
+
+    return (
+        <FloatingMessageCounterContainer
+            accessibilityHint={translate('accessibilityHints.scrollToNewestMessages')}
+            containerStyles={[styles.floatingMessageCounterTransformation(translateY)]}
+        >
+            <View style={styles.floatingMessageCounter}>
+                <View style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter]}>
+                    <Button
+                        success
+                        small
+                        onPress={props.onClick}
+                    >
+                        <View style={[styles.flexRow, styles.alignItemsCenter]}>
+                            <Icon
+                                small
+                                src={Expensicons.DownArrow}
+                                fill={themeColors.textLight}
+                            />
+                            <Text
+                                selectable={false}
+                                style={[styles.ml2, styles.buttonSmallText, styles.textWhite]}
+                            >
+                                {translate('newMessages')}
+                            </Text>
+                        </View>
+                    </Button>
                 </View>
-            </FloatingMessageCounterContainer>
-        );
-    }
+            </View>
+        </FloatingMessageCounterContainer>
+    );
 }
 
 FloatingMessageCounter.propTypes = propTypes;
 FloatingMessageCounter.defaultProps = defaultProps;
-
-export default withLocalize(FloatingMessageCounter);
+FloatingMessageCounter.displayName = 'FloatingMessageCounter';
+export default React.memo(FloatingMessageCounter);
```

Comment:
Why are we memoizing the component? Did you notice any performance issues?

Reply:
`FloatingMessageCounter` used to be `PureComopent`. Which, I believe that it prevents the components from re-rendering unless their own states or props are changed.

I used `React.memo` to have the same effect.

Reply:
thanks